### PR TITLE
[Scala] starting alpha support for Scala

### LINKED ISSUE #1953: Support for Scala
Scala is growing as a choice in high-performance computing environments and data science, offering both OO and functional approaches. Developers would highly benefit from support in semgrep.

### PR DESCRIPTION
**Pull request by aryx**
This closes https://github.com/returntocorp/semgrep/issues/1953

test plan:
test files included

PR checklist:
- [x] changelog is up to date

### CODE REVIEW

File: semgrep/semgrep/semgrep_types.py [hunk 1]
```diff
@@ -66,6 +67,7 @@ class Language_util:
         Language.RUST: [Language.RUST.value, "Rust", "rs"],
         Language.KOTLIN: [Language.KOTLIN.value, "Kotlin", "kotlin"],
         Language.YAML: [Language.YAML.value, "Yaml"],
+        Language.SCALA: [Language.SCALA.value, "scala"],
```
Comment (mschwager) [inline semgrep/semgrep/semgrep_types.py]: These two are the same thing :+1: 

```suggestion
        Language.SCALA: [Language.SCALA.value],
```

Comment (aryx) [reply semgrep/semgrep/semgrep_types.py]: oops, copy pasting without thinking

Comment (aryx) [reply semgrep/semgrep/semgrep_types.py]: I didn't know you could commit suggested changes. How do you suggest those changes? There's a button somewhere?

Comment (mschwager) [reply semgrep/semgrep/semgrep_types.py]: Yup, above the comment box with the little "+" and "-":

2021-05-21-100019_904x408_scrot

Comment (mschwager) [reply semgrep/semgrep/semgrep_types.py]: And you can drag the mouse over multiple lines to suggest multiple!

**Comment by mschwager** [comment]
Looks like Scala isn't hitting the parse rate requirements for alpha? https://dashboard.semgrep.dev/metric/semgrep.core.scala.parse.pct

EDIT: Should we go with experimental for now?

**Comment by mschwager** [comment]
> Looks like Scala isn't hitting the parse rate requirements for alpha? https://dashboard.semgrep.dev/metric/semgrep.core.scala.parse.pct

EDIT: Should we go with experimental for now?

**Comment by aryx** [comment]
Those stats are for tree-sitter-scala, that we don't use

**Comment by aryx** [comment]
But we need to wait for @mjambon to update the parsing stats to use tree-sitter or pfff, and especially to test parsing AND whether the conversion to the generic AST succeed.

**Comment by mjambon** [comment]
Feel free to remove tree-sitter-scala from ocaml-tree-sitter to reduce confusion and maintenance burden.

**Comment by aryx** [comment]
will do

**Comment by aryx** [comment]
done: https://github.com/returntocorp/ocaml-tree-sitter/pull/207

**Pull request merged**